climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+            - VIIRS: remove data where confidence values are set to low (keep
+                nominal and high values)
+
+        Parameters
+        ----------
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+
+        Returns
+        -------
+            firms : pd.DataFrame
+
+        """
+        if isinstance(csv_firms, pd.DataFrame):
+            firms = csv_firms
+        else:
+            firms = pd.read_csv(csv_firms)
+        # Check for the type of instrument (MODIS vs VIIRS)
+        # Remove data with low confidence interval
+        # Uniformize the name of the birghtness columns between VIIRS and MODIS
+        temp = pd.DataFrame()
+        if 'instrument' in firms.columns:
+            if firms.instrument.any() == 'MODIS' or firms.instrument.any() == 'VIIRS':
+                firms_modis = firms.drop(firms[firms.instrument == 'VIIRS'].index)
+                firms_modis.confidence = np.array(
+                    list(map(int, firms_modis.confidence.values.tolist())))
+                firms_modis = firms_modis.drop(firms_modis[ \
+                    firms_modis.confidence < CLEAN_THRESH].index)
+                temp = firms_modis
+                firms_viirs = firms.drop(firms[firms.instrument == 'MODIS'].index)
+                if firms_viirs.size:
+                    firms_viirs = firms_viirs.drop(firms_viirs[firms_viirs.confidence == 'l'].index)
+                    firms_viirs = firms_viirs.rename(columns={'bright_ti4':'brightness'})
+                    temp = temp.append(firms_viirs, sort=True)
+                    temp = temp.drop(columns=['bright_ti4'])
+
+            firms = temp
+            firms = firms.reset_index()
+            firms = firms.drop(columns=['index'])
+
+        firms['iter_ev'] = np.ones(len(firms), bool)
+        firms['cons_id'] = np.zeros(len(firms), int) - 1
+        firms['event_id'] = np.zeros(len(firms), int)
+        firms['clus_id'] = np.zeros(len(firms), int) - 1
+        firms['datenum'] = np.array(u_dt.str_to_date(firms['acq_date'].values))
+        return firms
+
+    @staticmethod
+    def _firms_resolution(firms):
+        """ Returns resolution of satellite used in FIRMS in degrees
+
+        Parameters
+        ----------
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+
+        Returns
+        -------
+            res_data/ONE_LAT_KM : float
+                resolution in degrees
+
+        """
+        # Resolution in km of the centroids depends on the data origin.
+        if 'instrument' in firms.columns:
+            if firms['instrument'].any() == 'MODIS':
+                res_data = 1.0
+            else:
+                res_data = 0.375 # For VIIRS data
+        else:
+            res_data = RES_DATA # For undefined data origin, defined by user
+        return res_data/ONE_LAT_KM
```

Comment:
Why divide by ONE_LAT_KM? This is quite obscure. Also, degrees and kilometers cannot just be converted like this, as this depends strongly on the latitude.

Reply:
I don't know the full reasoning behind, as this is not my code.
However, I think the reason why this is done, is because the satellite data comes with resolution information in [m]. Hence, in order to set the grid we should normalise this

Reply:
But this still makes not much sense. You cannot convert degrees in to kilometers this way since the convertion factor is latitude dependent. Please check.

Reply:
I think it's longitude dependent, right? That's why it's called ONE_*LAT*_KM. Latitude is, to a first approximation, quite consistent in terms of how many kms equal one degree _latitude_ . For longitude, there's more or less simple approximations on how to convert a degree longitude to kms, depending on where on the globe you are. we have a util func for this in coordinates, as far as i'm aware.

Reply:
But in general, yes, just assuming a constant conversion factor for longitude will lead to "ungenauigkeiten" that are hidden here.

Reply:
Sure! I meant, depending on the latitude, the longitude degree/km ratio changes.

Reply:
I'm not entirely sure if that makes a difference. I think it would only make a difference if the MODIS data in Australia is at a different resolution than the one closer to the equator. Else it's really just a step for normalisation, as the grid-spacing of the centroids is anyway held constant and not projected onto a sphere

Reply:
I am sorry, maybe I am missing something obvious, but I really do not understand this. Why then do this 'normalisation' if it is meaningless? Either it is important to have kilometers, in which case one cannot simply renormalize this way, in particular since wildfires are present on very and very latitudes too. Or, kilometers are not relevant, in which case no normalisation is required. 

And what do you mean by 'the grid-spacing of the centroids is anyway held constant' ? This completely depends on the chosen centroids... which depends on the coordinate systems...